File: map/lab2/src/domain/Student.java
package domain;

public class Student implements HasID<Integer> {
    private int id;
    private String name;
    private String group;
    private String email;
    private String guide;


    public Student(int _id, String _name, String _group, String _email, String _guide) {
        this.id = _id;
        this.name = _name;
        this.group = _group;
        this.email = _email;
        this.guide = _guide;
    }

    @Override
    public Integer getId() {
        return this.id;
    }

    public String getName() {
        return this.name;
    }

    public String getGroup() {
        return this.group;
    }

    public String getEmail() {
        return this.email;
    }

    public String getGuide() {
        return this.guide;
    }

    @Override
    public void setId(Integer _id) {
        this.id = _id;
    }

    public void setName(String _name) {
        this.name = _name;
    }

    public void setGroup(String _group) {
        this.group = _group;
    }

    public void setEmail(String _email) {
        this.email = _email;
    }

    public void setGuide(String _guide) {
        this.guide = _guide;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o)
            return true;

        if (o == null || getClass() != o.getClass())
            return false;

        Student student = (Student) o;

        if (id == student.getId() || student.name != null)
            return false;
        return true;
    }

    @Override
    public int hashCode() {
        return 0;
    }

    @Override
    public String toString() {
        return this.getId() + "; " + this.getName() + "; " + this.getGroup() + "; " + this.getEmail() + "; " + this.getGuide();
    }
}

File: map/lab7/lab7/Domain/Student.cs
﻿using System;
using lab7.Repository;

namespace lab7.Domain
{
    public class Student : IHasId<int>
    {
        private int _id;
        private string _name;
        private string _group;
        private string _email;
        private string _guide;

        public int Id
        {
            get { return _id; }
            set { _id = value; }
        }

        public string Name
        {
            get { return _name; }
            set { _name = value; }
        }

        public string Group
        {
            get { return _group; }
            set { _group = value; }
        }

        public string Email
        {
            get { return _email; }
            set { _email = value; }
        }

        public string Guide
        {
            get { return _guide; }
            set { _guide = value; }
        }

        public Student(int idP, string nameP, string groupP, string emailP, string guideP)
        {
            _id = idP;
            _name = nameP;
            _group = groupP;
            _email = emailP;
            _guide = guideP;
        }

        public override string ToString()
        {
            return "Id: " + _id + "\n" + "Name: " + _name + "\nGroup: " + _group + "\nEmail: " + _email + "\nGuide: " + _guide + "\n";
        }

        public override bool Equals(Object o)
        {
            if (this == o)
                return true;

            if (o == null || GetType().Name != o.GetType().Name)
                return false;

            Student student = (Student) o;

            if (_id != student.Id || student._name != null)
                return false;
            return true;
        }
    }
}